pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -366,12 +379,26 @@ func (ctrl *Controller) generateOriginalContainerRuntimeConfigs(role string) (*i
 }
 
 func (ctrl *Controller) syncStatusOnly(cfg *mcfgv1.ContainerRuntimeConfig, err error, args ...interface{}) error {
-	if cfg.GetGeneration() != cfg.Status.ObservedGeneration {
-		cfg.Status.ObservedGeneration = cfg.GetGeneration()
-		cfg.Status.Conditions = append(cfg.Status.Conditions, wrapErrorWithCondition(err, args...))
+	statusUpdateErr := retry.RetryOnConflict(updateBackoff, func() error {
+		if cfg.GetGeneration() != cfg.Status.ObservedGeneration {
+			cfg.Status.ObservedGeneration = cfg.GetGeneration()
+			cfg.Status.Conditions = append(cfg.Status.Conditions, wrapErrorWithCondition(err, args...))
+		} else if cfg.GetGeneration() == cfg.Status.ObservedGeneration && err == nil {
+			// If the CR was created before a matching label was added, the CR would be in failure state
+			// However the observed generation would be the same, so check if err is nil as well
+			// Which means that, the ctrcfg was finally successfully able to sync. In that case update the status
+			// to success and clear the previous failure status
+			cfg.Status.Conditions = []mcfgv1.ContainerRuntimeConfigCondition{wrapErrorWithCondition(err, args...)}
+		}
+		_, lerr := ctrl.client.MachineconfigurationV1().ContainerRuntimeConfigs().UpdateStatus(cfg)
+		return lerr
+	})
+	// If an error occured in updating the status just log it
+	if statusUpdateErr != nil {
+		glog.Warningf("error updating container runtime config status: %v", statusUpdateErr)
 	}
-	_, lerr := ctrl.client.MachineconfigurationV1().ContainerRuntimeConfigs().UpdateStatus(cfg)
-	return lerr
+	// Want to return the actual error received from the sync function
+	return err
```

Comment:
I think the `lerr` `err` is kind of confusing. Perhaps renaming the func argument `err` to something like syncErr and renaming `lerr` -> `err` would be a little more clear?

Reply:
Renamed it to `updateErr`, hopefully that is more clear.

Reply:
thanks this is perfect!